### ISSUE TITLE
Use PyTorch Lightning to handle the training (free checkpointing + logging + 16-bit precision)

### ISSUE THREAD

**Issue by williamFalcon** [karpathy/minGPT#11]
Awesome repo!

However, not sure why go through the effort of implementing your own trainer again...

In lightning we already support:

- automatic checkpoint loading/saving
- multi-cpu
- multip-gpu
- multi-tpu core
- 16-bit precision (amp and native)
- accumulated gradients
- and about 40+ more features.

Not to mention it's maintained by a team of over 20+ fulltime engineers and 200+ open-source contributors and has been adopted by over 400 companies and research labs.

https://pytorch-lightning.readthedocs.io/en/latest/new-project.html

**Comment by ykumards** [comment]
![tenor](https://user-images.githubusercontent.com/5177126/90639740-cab29000-e237-11ea-9482-59dca0d8a95c.gif)
**min**GPT, no? 😬

**Comment by karpathy** [comment]
@williamFalcon your link above doesn't work btw. Closing the Issue and will look at the PR shortly.

**Comment by williamFalcon** [comment]
sent the old link

https://pytorch-lightning.readthedocs.io/en/latest/3_steps.html
